### ISSUE TITLE
Punycode library needs updating

### ISSUE THREAD

**Issue by brianteeman** [joomla/joomla-cms#9497]
We are using v1.2.4 current version is 1.4.1
 https://github.com/bestiejs/punycode.js

**Comment by wilsonge** [comment]
You want to give it a go - it *should* be as simple as replacing the two files with the ones from the github repo there :)

**Comment by brianteeman** [comment]
I tried and messed up my repo. No idea why. Can try again tomorrow I guess
but it will take a while to fix the repo

**Comment by infograf768** [comment]
There are 2 files to replace, the minified one which has to be renamed to uncompressed, and the compressed one.
also I guess it has to be tested. I just have to remember when we use it... :cactus:

**Comment by brianteeman** [comment]
When 3.5 is released I will try to rebuild my repo and do the PR unless
someone has done it first

On 21 March 2016 at 11:24, infograf768 <notifications@github.com> wrote:

> There are 2 files to replace, the minified one which has to be renamed to
> uncompressed, and the compressed one.
> also I guess it has to be tested. I just have to remember when we use
> it... [image: :cactus:]
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/issues/9497#issuecomment-199232430>
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by infograf768** [comment]
I now found out what it is used for (email validation)

See https://github.com/joomla/joomla-cms/pull/2788

**Comment by infograf768** [comment]
We will have to test a change we did in https://github.com/joomla/joomla-cms/pull/3429
```
-				value = punycode.encode(value);
+				value = punycode.toASCII(value);
```

in validate.js 
as there were some changes in punycode.js

**Comment by brianteeman** [comment]
Closed see #9889 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9497">issues.joomla.org/joomla-cms/9497</a>.</sub>
